app/setupBackground.js

```diff
@@ -5,52 +5,61 @@ import { isRunningUnpacked, isRunningDevelopment } from 'appConstants';
 
 const BACKGROUND_PROCESS = path.join( __dirname, 'bg.html' );
 let backgroundProcess = null;
-
-const setupBackground = () =>
-{
+// NOTE: Setting up function as promise in a failed attempt to make sure it runs before `openWindow()` is called in main.js
+const setupBackground = async ( ) => new Promise( ( resolve, reject ) => {
     logger.info( 'Setting up Background Process', backgroundProcess );
 
     if ( backgroundProcess === null )
     {
         logger.info('loading:', BACKGROUND_PROCESS );
 
         backgroundProcess = new BrowserWindow( {
-            width: 300,
-            height: 450,
-            show: false,
-            frame: false,
-            fullscreenable: false,
-            resizable: false,
-            transparent: true,
+            width          : 300,
+            height         : 450,
+            show           : false,
+            frame          : false,
+            fullscreenable : false,
+            resizable      : false,
+            transparent    : true,
             webPreferences : {
                 // partition               : 'persist:safe-tab', // TODO make safe?
-                nodeIntegration         : true,
+                nodeIntegration     : true,
                 // Prevents renderer process code from not running when window is hidden
                 backgroundThrottling: false
             }
         } );
 
         // Hide the window when it loses focus
-        backgroundProcess.on('blur', () => {
-          if (!backgroundProcess.webContents.isDevToolsOpened()) {
-            backgroundProcess.hide()
-          }
-      });
+        backgroundProcess.on( 'blur', () =>
+        {
+            if ( !backgroundProcess.webContents.isDevToolsOpened() )
+            {
+                backgroundProcess.hide();
+            }
+        } );
 
         backgroundProcess.webContents.on( 'did-finish-load', () =>
         {
             logger.verbose( 'Background process renderer loaded.');
 
-            if( isRunningUnpacked || isRunningDevelopment )
+            if ( isRunningUnpacked || isRunningDevelopment )
             {
-                backgroundProcess.webContents.openDevTools()
+                backgroundProcess.webContents.openDevTools();
             }
+            resolve( backgroundProcess );
+        } );
+
+        backgroundProcess.webContents.on( 'did-fail-load', ( event, code, message ) =>
+        {
+            reject( message );
         } );
 
         backgroundProcess.loadURL( `file://${BACKGROUND_PROCESS}` );
     }
-
-    return backgroundProcess;
-};
+    else
+    {
+        resolve( backgroundProcess );
```

Comment:
i think you want this inside of the `did-finish-load` callback above, @hunterlester